crates/nu-parser/src/parser.rs

```diff
@@ -592,7 +592,14 @@ fn calculate_end_span(
             } else if positional_idx < signature.required_positional.len()
                 && spans.len() > (signature.required_positional.len() - positional_idx)
             {
-                spans.len() - (signature.required_positional.len() - positional_idx - 1)
+                spans.len()
+                    - (signature.required_positional.len()
+                        - positional_idx
+                        - (if signature.category == Category::Core {
+                            1
+                        } else {
+                            2
```

Comment:
I don't think we can make this general assumptions that all commands that get labelled with `Category::Core` would share this behavior and should rather fix where their parsing is misbehaving.

Reply:
The mistake is the whole signature. Can manipulate the core command signature on positional argument .

Reply:
The second condition should be disabled for default command.

Reply:
From this change it is really hard to follow how it relates to custom command calls failing during syntax highlighting with more than a certain number of positional arguments but it not being a problem for lower numbers. As described by #9072 

Based on the diff it just appears to add another special case for which it is unclear why it needs to be there that will make future bugfixes an even greater nightmare.

To close #9072 we should also include a test that demonstrates the parser doesn't panic in this circumstance.

Reply:
Please attention to the issue ,
When you insert a args on cli it panics , not when the whole command executes . How can I add tests with it then? 
Review the signature some command like `if` , in if there are multi span which are greater than size of positional arg , but when I create function `def` it attach a signature which has span_idx that only should increase when it wants to calculate the index.